public/app/core/utils/CorsSharedWorker.ts

```diff
@@ -0,0 +1,39 @@
+// Almost identical to CorsWorker.ts. Main difference being it allows loading a SharedWorker if browser supports it
+
+export function sharedWorkersSupported() {
+  return typeof window.SharedWorker !== 'undefined';
+}
+
+/**
+ * The base class is created dynamically here to allow later on syntax like:
+ * import { CorsSharedWorker as SharedWorker } from '../utils/CorsSharedWorker';
+ * And also take into account cases where SharedWorked is not available in the browser (mainly mobile browsers)
+ *
+ * It's important to use: new SharedWorker(...) syntax instead of new CorsSharedWorker(...) due to how web-workers
+ * are processing workers syntax (more details https://webpack.js.org/guides/web-workers/)
+ */
+const BaseSharedWorkerClass = sharedWorkersSupported() ? window.SharedWorker : null;
+
+// @ts-ignore
+export class CorsSharedWorker extends BaseSharedWorkerClass {
```

Comment:
`null` is not a valid constructor, and typescript is telling you this. This PR needs to not have `ts-ignore` in it.

Reply:
👍  removed `// @ts-ignore` by implementing what @svennergr suggested above